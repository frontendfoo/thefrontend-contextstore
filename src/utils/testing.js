import React, { Fragment } from 'react'
import TestRenderer from 'react-test-renderer'

export const testHook = (customHook, Provider = Fragment, initialState) => {
  let hook

  TestRenderer.create(
    <Provider {...(initialState && { initialState })}>
      {React.createElement(() => {
        hook = customHook()
        return null
      })}
    </Provider>
  )

  return hook
}
