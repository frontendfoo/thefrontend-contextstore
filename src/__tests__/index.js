import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import TestRenderer from 'react-test-renderer'
import { testHook } from '../utils/testing'
import makeStore from '../'

const { act } = TestRenderer
const renderer = new ShallowRenderer()

describe('The makeStore script', () => {
  describe('The Provider Component', () => {
    it('should match its snapshot', () => {
      const { Provider } = makeStore()
      const render = renderer.render(<Provider>Children</Provider>)
      expect(render).toMatchSnapshot()
    })

    it('should match its snapshot with initialState', () => {
      const { Provider } = makeStore({ test: 'yes' })
      const render = renderer.render(<Provider>Children</Provider>)
      expect(render).toMatchSnapshot()
    })

    it('should match its snapshot with initialState prop', () => {
      const { Provider } = makeStore()
      const render = renderer.render(
        <Provider initialState={{ test: 'yes' }}>Children</Provider>
      )
      expect(render).toMatchSnapshot()
    })

    it('should match its snapshot with initialState and initialState prop merge', () => {
      const { Provider } = makeStore({ test1: 'yes' })
      const render = renderer.render(
        <Provider initialState={{ test2: 'no' }}>Children</Provider>
      )
      expect(render).toMatchSnapshot()
    })
  })

  describe('The useStore hook', () => {
    const { Provider, useStore } = makeStore({ test: 'test state' })
    let hook

    const callHook = () =>
      testHook(() => {
        hook = useStore()
      }, Provider)

    beforeEach(() => {
      callHook()
    })

    it('should have a state Prop', () => {
      expect(hook.state).toEqual({ test: 'test state' })
    })

    it('should update state using setState', () => {
      act(() => {
        hook.setState({ test: 'new state' })
      })
      expect(hook.state).toEqual({ test: 'new state' })
    })
  })
})
