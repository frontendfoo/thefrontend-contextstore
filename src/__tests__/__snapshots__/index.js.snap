// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The makeStore script The Provider Component should match its snapshot 1`] = `
<Context.Provider
  value={
    Array [
      Object {},
      [Function],
    ]
  }
>
  Children
</Context.Provider>
`;

exports[`The makeStore script The Provider Component should match its snapshot with initialState 1`] = `
<Context.Provider
  value={
    Array [
      Object {
        "test": "yes",
      },
      [Function],
    ]
  }
>
  Children
</Context.Provider>
`;

exports[`The makeStore script The Provider Component should match its snapshot with initialState and initialState prop merge 1`] = `
<Context.Provider
  value={
    Array [
      Object {
        "test1": "yes",
        "test2": "no",
      },
      [Function],
    ]
  }
>
  Children
</Context.Provider>
`;

exports[`The makeStore script The Provider Component should match its snapshot with initialState prop 1`] = `
<Context.Provider
  value={
    Array [
      Object {
        "test": "yes",
      },
      [Function],
    ]
  }
>
  Children
</Context.Provider>
`;
